### DISCUSSION TITLE
Row level security

### DISCUSSION THREAD

**Discussion by nephix** [prisma/prisma#20168] [Q&A] [answered]
In the row level security example on https://github.com/prisma/prisma-client-extensions/tree/main/row-level-security it says:

> This extension is provided as an example only. It is not intended to be used in production environments.

But no mention of "why". 

Is the approach not good in production? If yes, why?
Does the example miss common "production features" like input validation etc?

**Comment by nurul3101** [comment, marked as answer]
Hello @nephix 👋 

Thanks for pointing this out, I believe the intention behind the statement was that because Prisma Client Extensions were in preview when the examples were added, we didn't recommend it for production use.

But since version `4.16.0`, [Client Extensions](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions) are Generally Available (GA), so they can be used in production. I'll highlight this so that our descriptions can be updated now.

**Reply by nurul3101** [reply]
Hey @nephix 👋 

I had a discussion internally about the warning. The intention behind the statement in the RLS example was that these extensions are supposed to be treated as examples. We would recommend you actually verify that everything is working as expected before you use it in production, considering that it's related to security.

Client Extensions are in GA and are super stable now, these examples serves as a reference and you should make sure that they fully serve your needs and you test them before moving it to production.

**Reply by nephix** [reply]
Thanks @nurul3101

**Reply by nurul3101** [reply]
No worries :)
